imap/jmap_mail.c

```diff
@@ -12313,7 +12322,8 @@ static int _email_bulkupdate_plan_keywords(struct email_bulkupdate *bulk, ptrarr
     int i;
 
     /* Open seen.db, if required */
-    if (strcmp(bulk->req->accountid, bulk->req->userid)) {
+    if (bulk->from_account_id ||  // always need seen.db for Email/copy
```

Comment:
Did you deliberately keep this nullable? I figure it might be simpler to reason about if its always set but might differ from req->accountid. note that req->accountid also always is set.

Reply:
I'm not sure I follow what you're saying. For /copy, accountid and from_accountid will always differ but accountid and userid might be the same, which is why I added the check for from_accountid. 

The alternative would be to open seen.db in email_copy_bulk() if needed and check if it's already open in exec_keywords

Reply:
I mean: why can `bulk->from_account_id` either be NULL or some C string value. Can't it be always set to the accountid from which the messages are copied? Just as req->accountid always is set to some C-string, it might be helpful to also know that bulk->from_account_id is set to some value.

Reply:
from_account_id being set is the indicator that we're doing a /copy. If it's not set then we're doing /set{update}. If you think it needs clarification, I can add an is_copy field to struct bulkupdate and use that for triggering the copy logic

Reply:
yes, I'd prefer either that or rename `from_account_id` to `copy_from_account_id`